### PR TITLE
Fixed component discriminator usability issue #87

### PR DESCRIPTION
**Pull request by NellyHargeliya**

### CODE REVIEW

**Comment by NellyHargeliya** [comment]
When requesting user input for parameters with the same name but different component discriminators, the message will now include information about the component to which the parameter belongs. This ensures clarity and avoids confusion when multiple parameters share the same name but have different component discriminators.
Added:
- Condition for displaying a message about the component to which the parameter belongs, only if the $ENV variable is not empty.
 Sample message, 
 _"* Setting parameter ingress.hosts **KUBEFLOW**_
     _Enter value (6jy54gla452z74fsp9e99csj292548m... #random): "_
- Tests for this feature

**Review by oginskis — COMMENTED** [review]
I believe it should display the component's name, not the environment variable's name.

For example, if I have:

```
.....
- name: kubernetes.namespace
  component: nginx
  fromEnv: NGINX_NAMESPACE
.....
```

I expect it output something like:

```
* Setting parameter kubernetes.namespace nginx
  Enter value (a5rnzq55sklea3gksah8egv2vqvg3fo... #random):
```
not
```
* Setting parameter kubernetes.namespace NGINX_NAMESPACE
  Enter value (a5rnzq55sklea3gksah8egv2vqvg3fo... #random):
```

OR you agreed with @akranga that it should be the name of `env` var?

File: tests/ask.bats [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  run  ask env "BATS_TEST_HOST" -m "parameter test.hosts" +empty -d ".ask.env" -d ".env" -ask-env
+  assert_success
+  assert_output --partial "* Setting parameter test.hosts BATS_TEST_HOST"
+  assert_output --partial "BATS_TEST_HOST already set"
+}
+
+teardown_file() {
+  rm -fv "$DOTENV_FILE" ask.*.yaml
+  rm .env .ask.env
+}
```
Comment (oginskis) [inline tests/ask.bats]: Empty line at the end is missing

File: env/configure [hunk 2]
```diff
@@ -69,5 +69,6 @@ fi
 
 for e in $(params listenv); do
   pname="$(params "$e" envvar | jq -cMr '.name? | select(.)')"
-  ask env "$e" -m "parameter $pname" +empty -d "$DOT_ENV" -d "$(files abspath ".env")" -ask-env
+  pcomponent="$(params "$e" envvar | jq -cMr '.component? | select(.)')"
+  ask env "$e" -m "parameter $pname for $pcomponent" +empty -d "$DOT_ENV" -d "$(files abspath ".env")" -ask-env
```
Comment (oginskis) [inline env/configure]: If $pcomponent is empty, do not write "for" in the message

File: bin/ask [hunk 3]
```diff
@@ -220,6 +220,8 @@ fi
 
 if test -z "$message"; then
   message="$ENV variable"
+elif test -n "$ENV"; then
+  message="$message $ENV"
```
Comment (oginskis) [inline bin/ask]: add "," after $message

**Review by oginskis — COMMENTED** [review]
Please address my comments

**Comment by oginskis** [comment]
@NellyHargeliya any updates? Let's get this merged in

**Comment by NellyHargeliya** [comment]
@oginskis updated according to your comments

**Comment by akranga** [comment]
I would suggest `Setting parameter kubernetes.namespace (component: nginx)` here...

**Pull request merged**